src/core/contracts/identity.test.ts

```diff
@@ -137,6 +142,24 @@ describe("identity", () => {
       expect(ethers.utils.isAddress(address)).toBeTruthy();
     });
   });
+
+  describe("#upsertDelegate", () => {
+    let contractAddress: EthereumAddress;
+    let contractOwner: EthereumAddress;
+
+    beforeAll(async () => {
+      contractOwner = await signer.getAddress();
+      const identityContract = await new Identity__factory(signer).deploy(contractOwner);
+      await identityContract.deployed();
+      contractAddress = identityContract.address;
+    });
+
+    it("upserts delegate", async () => {
+      await upsertDelegate(contractAddress, NON_OWNER, 0x1);
+
+      expect(await isAuthorizedTo(NON_OWNER, contractAddress, 2, 60)).toBeTruthy();
```

Comment:
This should really have an update as well as an insert test. It would be ok to add a second call to this test with a second expectation.